cloud/scope/machine.go

```diff
@@ -766,6 +771,99 @@ func (m *MachineScope) DeleteVPCLoadBalancerPoolMember() error {
+		}
+		// Verify the instance has a primary network interface IP address needed to delete the correct Load Balancer Pool Member.
+		if instanceDetails.PrimaryNetworkInterface == nil || instanceDetails.PrimaryNetworkInterface.PrimaryIP == nil || instanceDetails.PrimaryNetworkInterface.PrimaryIP.Address == nil {
+			return fmt.Errorf("error instance is missing the primary network interface IP address for load balancer pool member deletion for machine: %s", m.IBMVPCMachine.Name)
+		}
+		m.Logger.V(5).Info("collected instance details for load balancer pool member deletion", "machienName", m.IBMVPCMachine.Name, "instanceID", *instanceDetails.ID, "instanceIP", *instanceDetails.PrimaryNetworkInterface.PrimaryIP.Address)
+
+		// Retrieve the Load Balancer.
+		var loadBalancerDetails *vpcv1.LoadBalancer
+		if member.LoadBalancer.ID != nil {
+			loadBalancerDetails, _, err = m.IBMVPCClient.GetLoadBalancer(&vpcv1.GetLoadBalancerOptions{
+				ID: member.LoadBalancer.ID,
+			})
+		} else if member.LoadBalancer.Name != nil {
+			loadBalancerDetails, err = m.IBMVPCClient.GetLoadBalancerByName(*member.LoadBalancer.Name)
+		} else {
+			return fmt.Errorf("error load balancer has no id or name for load balancer pool member deleteion")
+		}
+		if err != nil {
+			return fmt.Errorf("error retrieving load balancer for load balancer pool member deletion for machine %s: %w", m.IBMVPCMachine.Name, err)
+		}
+		m.Logger.V(5).Info("collected load balancer for load balancer pool member deletion", "machineName", m.IBMVPCMachine.Name, "loadBalancerID", *loadBalancerDetails.ID)
+
+		// Lookup the Load Balancer Pool ID, if only name is available.
+		loadBalancerPoolID, err := m.getLoadBalancerPoolID(ptr.To(member.Pool), *loadBalancerDetails.ID)
+		if err != nil {
+			return fmt.Errorf("error retrieving load balancer pool id for load balancer pool member deletion for machine %s: %w", m.IBMVPCMachine.Name, err)
+		}
+		m.Logger.V(5).Info("collected load balancer pool id for load balancer pool member deletion", "machineName", m.IBMVPCMachine.Name, "loadBalancerPoolID", *loadBalancerPoolID)
+
+		listMembersOptions := &vpcv1.ListLoadBalancerPoolMembersOptions{
+			LoadBalancerID: loadBalancerDetails.ID,
+			PoolID:         loadBalancerPoolID,
+		}
+
+		m.Logger.V(5).Info("list load balancer pool members options", "machineName", m.IBMVPCMachine.Name, "options", *listMembersOptions)
+		poolMembers, _, err := m.IBMVPCClient.ListLoadBalancerPoolMembers(listMembersOptions)
+		if err != nil {
+			return fmt.Errorf("error retrieving load balancer pool members for load balancer pool member deletion for machine %s: %w", m.IBMVPCMachine.Name, err)
+		}
+
+		for _, poolMember := range poolMembers.Members {
+			poolMemberTarget, ok := poolMember.Target.(*vpcv1.LoadBalancerPoolMemberTarget)
+			if !ok {
+				continue
+			}
+			if poolMemberTarget.Address != nil && *poolMemberTarget.Address == *instanceDetails.PrimaryNetworkInterface.PrimaryIP.Address {
```

Comment:
```suggestion
			if poolMemberTarget.Address != nil && *poolMemberTarget.Address != *instanceDetails.PrimaryNetworkInterface.PrimaryIP.Address {
			continue
			}
```

Reply:
Ah, will fix

Reply:
I've refactored and combined with previous if block to `continue`.